pyccl/tk3d.py

```diff
@@ -213,3 +216,153 @@ def get_spline_arrays(self):
+        deltah (float): the variation of h to compute T_{h}(k) by
+            the two-sided numerical derivative method.
+        a_arr (array): an array holding values of the scale factor
+            at which the trispectrum should be calculated for
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        lk_arr (array): an array holding values of the natural
+            logarithm of the wavenumber (in units of Mpc^-1) at
+            which the trispectrum should be calculated for
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        extrap_order_lok (int): extrapolation order to be used on
+            k-values below the minimum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        extrap_order_hik (int): extrapolation order to be used on
+            k-values above the maximum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        use_log (bool): if `True`, the trispectrum will be
+            interpolated in log-space (unless negative or
+            zero values are found).
+
+    Returns:
+        :class:`~pyccl.tk3d.Tk3D`: SSC effective trispectrum.
+    """
+
+    if lk_arr is None:
+        status = 0
+        nk = lib.get_pk_spline_nk(cosmo.cosmo)
+        lk_arr, status = lib.get_pk_spline_lk(cosmo.cosmo, nk, status)
+        check(status, cosmo=cosmo)
+    if a_arr is None:
+        status = 0
+        na = lib.get_pk_spline_na(cosmo.cosmo)
+        a_arr, status = lib.get_pk_spline_a(cosmo.cosmo, na, status)
+        check(status, cosmo=cosmo)
+
+    k_use = np.exp(lk_arr)
+
+    Omega_c = cosmo["Omega_c"]
+    Omega_b = cosmo["Omega_b"]
+    h = cosmo["h"]
+    n_s = cosmo["n_s"]
+    A_s = cosmo["A_s"]
+    
+    extra_parameters = {"camb": {"halofit_version": "original",
+ }}
+    #set h-modified cosmology to take finite differencing
+    hp = h + deltah 
+    Omega_c_p = np.power((h/hp),2) * Omega_c #\Omega_c h^2 is fixed
+    Omega_b_p = np.power((h/hp),2) * Omega_b #\Omega_b h^2 is fixed
+
+    hm = h - deltah 
+    Omega_c_m = np.power((h/hm),2) * Omega_c #\Omega_c h^2 is fixed
+    Omega_b_m = np.power((h/hm),2) * Omega_b #\Omega_b h^2 is fixed
+
+    cosmo_hp = core.Cosmology(Omega_c=Omega_c_p, Omega_b=Omega_b_p,
+                            h=hp, n_s=n_s, A_s=A_s,
+                            transfer_function="boltzmann_camb",
+                            matter_power_spectrum="camb",
+                            extra_parameters=extra_parameters)
+    
+    cosmo_hm = core.Cosmology(Omega_c=Omega_c_m, Omega_b=Omega_b_m,
+                            h=hm, n_s=n_s, A_s=A_s,
+                            transfer_function="boltzmann_camb",
+                            matter_power_spectrum="camb",
+                            extra_parameters=extra_parameters)
+    
+    # Growth factor
+    Dp = background.growth_factor_unnorm(cosmo_hp,a_arr)
```

Comment:
You can remove the `background` dependency and use directly `cosmo_hp.growht_factor_unnorm(a_arr)`. Same in the next line

Reply:
Thank you for pointing this out. I have removed the background dependency as you mentioned.